rmgpy/molecule/group.py

```diff
@@ -1137,17 +1144,24 @@ def updateFingerprint(self):
                 isNitrogen = atomType.equivalent(nitrogen)
                 isOxygen   = atomType.equivalent(oxygen)
                 isSulfur   = atomType.equivalent(sulfur)
-                if isCarbon and not isNitrogen and not isOxygen and not isSulfur:
-                    self.carbonCount += 1
-                elif isNitrogen and not isCarbon and not isOxygen and not isSulfur:
-                    self.nitrogenCount += 1
-                elif isOxygen and not isCarbon and not isNitrogen and not isSulfur:
-                    self.oxygenCount += 1
-                elif isSulfur and not isCarbon and not isNitrogen and not isOxygen:
-                    self.sulfurCount += 1
-            if len(atom.radicalElectrons) == 1:
-                radical = atom.radicalElectrons[0]
-                self.radicalCount += radical
+                isChlorine = atomType.equivalent(chlorine)
+                isSilicon  = atomType.equivalent(silicon)
+                sum_is_atom = isCarbon + isNitrogen + isOxygen + isSulfur + isChlorine + isSilicon
+                if sum_is_atom == 1:
```

Comment:
What happens if sum_is_atom != 1? I can't imagine this happening, but maybe I am missing a subtlety here. I would think that this would either always evaluate to `True` (and thus it is not needed) or we need some sort of error handling if it does occur.

Reply:
I think this is related to the previous comment: an "atom" in a Group could be defined as `1 [C,N,O] ux px cx {2,S}`. In this case, atomType = [C,N,O], and `.equivalent` will return `True` for more than one isAtom. In this case, `sum_is_atom` will be 3.
I agree that it seems unnecessary (now that I rethink it) since it's already under the `if len(atom.atomType) == 1:` condition discussed above. We could either keep it to keep the same functionality as before in case I'm also missing a subtlety, or remove. I'll check.